use uuid.v4 for id generation and fix web benchmark

martinemmert opened the pull request:
Using the [uuid](https://www.npmjs.com/package/uuid) package and the method `v4` instead of the nodejs libraries `Buffer` and `crypto` to generate the id prefixes provides several benefits:

- the library is browser compatible without the need for any additional polyfills
- removes the need to configure bundlers to use polyfills
- reduces the file size of the web benchmark by nearly 95,47% 
- faster in all benchmarks

Relates to: #46 

Note: The Benchmark results in the docs are the ones before using uuid as id prefix generator

---

# Benchmark Results

## Environment

- 2.9 GHz 6-Core Intel Core i9 2018 MacBook Pro
- macOS Catalina 10.15.7
- Chrome Version 86.0.4240.111 (Offizieller Build) (x86_64)
- Node 14.8.0

## Nodejs benchmark

nodejs benchmark beforegit

Creating and destroying 50000 entities...
Create 50,000 entities with two simple components : 292.88ms
Changing the values of each component             : 5.94ms
Destroy 50,000 entities with two simple components: 299.73ms
Recreating components now that pool is established: 260.39ms

nodejs benchmark after

Creating and destroying 50000 entities...
Create 50,000 entities with two simple components : 189.27ms
Changing the values of each component             : 5.33ms
Destroy 50,000 entities with two simple components: 182.19ms
Recreating components now that pool is established: 157.21ms

---

## Web Benchmark

**browser benchmark before:**

Creating and destroying 500000 entities...
Create 50,000 entities with two simple components : 2925.81ms
Changing the values of each component             : 13.87ms
Destroy 50,000 entities with two simple components: 2641.12ms
Recreating components now that pool is established: 2434.85ms

**browser benchmark after:**

Creating and destroying 500000 entities...
Create 50,000 entities with two simple components : 1493.25ms
Changing the values of each component             : 11.41ms
Destroy 50,000 entities with two simple components: 1242.57ms
Recreating components now that pool is established: 1082.49ms

---

## Package Size

package size webbenchmark before: **861K**
package size webbenchmark after: **39K**

fritzy commented:
Looks like the package-lock.json wasn't updated in this pull request, causing the builds travis builds to fail.
Please correct this and I'll take a look.

Thanks a ton!

fritzy commented:
Ah, it looks like it's because the lock file is in conflict. So nevermind on that. I was able to test it.

Can you clean up the DOS newlines (^M) in your changes over the Overview.md file? I'll accept the changes and we'll call this 1.1.0 :)

martinemmert commented:
Strange, the DOS newlines might be something that happened during C&P 🤔 

Anyways, I fixed it and rebased my changes onto your last commit to the master branch.
